src/clients/SpokePoolClient.ts

```diff
@@ -822,14 +903,22 @@ export class SpokePoolClient extends BaseAbstractClient {
       });
     }
 
-    const deposits = depositEvents.map(({ args, blockNumber }) => {
+    const deposits = depositEvents.map((event) => {
+      let inputToken: string, inputAmount: BigNumber;
+
+      if (this.isV3DepositEvent(event)) {
+        ({ inputToken, inputAmount } = event.args);
+      } else {
+        // Coerce v2 deposit objects into v3 format.
```

Comment:
This needs a dedicated test to ensure that v2 deposits and v3 deposits are handled the same by batchComputeRealizezdLpFeePct(), and also that the translation into a v3 object is handled correctly.